src/Objs/Redirect/RedirectComponent.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react'
+import {
+  ensureUserIsLoggedIn,
+  isUserLoggedIn,
+  provideComponent,
+  urlFor,
+} from 'scrivito'
+import { InPlaceEditingPlaceholder } from '../../Components/InPlaceEditingPlaceholder'
+import { Redirect } from './RedirectObjClass'
+
+provideComponent(Redirect, ({ page }) => {
+  const requireLogin = page.get('requireLogin')
+  const link = page.get('link')
+  const url = link && urlFor(link)
+
+  React.useEffect(() => {
+    if (requireLogin && !isUserLoggedIn) {
+      ensureUserIsLoggedIn()
+      return
+    }
+
+    if (!link || !url) return
+
+    if (link.isExternal() && window.top) {
+      window.top.location.replace(url)
+    } else {
+      window.location.replace(url)
+    }
+  }, [requireLogin, link, url])
+
+  if (!link) {
+    return (
+      <InPlaceEditingPlaceholder center>
+        Select a link in the redirect page properties.
+      </InPlaceEditingPlaceholder>
+    )
+  }
+
+  return null
```

Comment:
I think it would be good to show a note for `isEditorLoggedIn()` and _not_ redirect - similar to the `Dropdown` obj component (e.g. https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://markus-protects-content.scrivito-portal-app.pages.dev/about-us-9db36de70dd4d157?_scrivito_workspace_id=k11311614e5c25b5&_scrivito_display_mode=view).

Reply:
Ok. But I think it doesn't make a big difference compared to `InPlaceEditingPlaceholder` - apart from the nicer format. Is that where you're coming from?

Reply:
After the changes in 87364a7 this is still not optimal. In the UI the redirects should never be followed but always be informative only.